modules/server/src/main/scala/seqexec/server/keywords/GdsClient.scala

```diff
@@ -56,142 +57,60 @@ object GdsClient {
-      <methodCall>
-        <methodName>HeaderReceiver.openObservation</methodName>
-        <params>
-          <param>
-            <value>
-              <string>{obsId.format}</string>
-            </value>
-          </param>
-          <param>
-            <value>
-              <string>{id}</string>
-            </value>
-          </param>
-          {keywordsParam(ks)}
-        </params>
-      </methodCall>
+    override def setKeywords(id: ImageFileId, ks: KeywordBag): F[Unit] =
+      makeRequest("keywords", KeywordRequest(id, ks).asJson)
 
     override def openObservation(
       obsId: Observation.Id,
       id:    ImageFileId,
       ks:    KeywordBag
-    ): F[Unit] = {
-      // Build the request
-      val xmlRpc      = openObservationRPC(obsId, id, ks)
-      val postRequest = POST(xmlRpc, gdsUri)
+    ): F[Unit] =
+      makeRequest("open-observation", OpenObservationRequest(obsId, id, ks).asJson)
 
-      // Do the request
-      client
-        .expect[Elem](postRequest)(scalaxml.xml)
-        .map(GdsClient.parseError)
-        .ensureOr(toSeqexecFailure)(_.isRight)
-        .void
-    }
+    override def closeObservation(id: ImageFileId): F[Unit] =
+      makeRequest("close-observation", CloseObservationRequest(id).asJson)
 
-    // Build an xml rpc request to close an observation
-    private def closeObservationRPC(id: ImageFileId): Elem =
-      <methodCall>
-        <methodName>HeaderReceiver.closeObservation</methodName>
-        <params>
-          <param>
-            <value>
-              <string>{id}</string>
-            </value>
-          </param>
-        </params>
-      </methodCall>
-
-    override def closeObservation(id: ImageFileId): F[Unit] = {
-      // Build the request
-      val xmlRpc      = closeObservationRPC(id)
-      val postRequest = POST(xmlRpc, gdsUri)
+    private def makeRequest(path: String, body: Json): F[Unit] = {
+      val uri         = gdsUri / path
+      val postRequest = POST(body, uri)
 
       // Do the request
       client
-        .expect[Elem](postRequest)(scalaxml.xml)
-        .map(GdsClient.parseError)
-        .ensureOr(toSeqexecFailure)(_.isRight)
+        .expect[String](postRequest)
+        .adaptErr { case e => SeqexecFailure.GdsException(e, uri) }
         .void
     }
+  }
 
-    private def keywordsParam(ks: KeywordBag): Elem =
-      <param>
-        <value>
-          <array>
-            <data>
-              {
-        ks.keywords.map { k =>
-          <value><string>{
-            s"${k.name},${KeywordType.gdsKeywordType(k.keywordType)},${k.value}"
-          }</string></value>
-        }
-      }
-            </data>
-          </array>
-        </value>
-      </param>
+  case class KeywordRequest(id: String, ks: KeywordBag)
+  case class OpenObservationRequest(obsId: Observation.Id, id: String, ks: KeywordBag)
+  case class CloseObservationRequest(id: String)
```

Comment:
Maybe we should add an abort message too, When the seqexec aborts we tell GDS to release the keywords it has in memory. We'd still need to do garbage collection on GDS but that would make it a bit cleaner

The DHS side doesn't have that so we may need to ignore it there.